### ISSUE TITLE
Support for Description Lists

### ISSUE THREAD

**Issue by ayosec** [pulldown-cmark/pulldown-cmark#67]
Some parsers have an extension to support description/definition lists. In https://talk.commonmark.org/t/description-list/289 there is a proposal to add it to CommonMark. A common syntax is to put a colon at the beginning of the line.

An example from the [Pandoc manual](http://pandoc.org/MANUAL.html#definition-lists):

```
Term 1

:   Definition 1

Term 2 with *inline markup*

:   Definition 2

        { some code, part of Definition 2 }

    Third paragraph of definition 2.
```

**Comment by darkdarkfruit** [comment]
That will be better if being supported

**Comment by ngirard** [comment]
This issue should be considered critical.

Since [mdBook](https://github.com/rust-lang-nursery/mdBook) relies on pulldown-cmark to produce its documentation, the current state of affairs is that **Rust projects just cannot use definition lists within their documentations**, unless they dismiss mdBook.

@ayosec has already given the 2 links that really matters here:
- the [discussion related to adding description lists to CommonMark as an extension](https://talk.commonmark.org/t/description-list/289) is ongoing since 2014, and probably not about to come to an end any time soon;
- the [Pandoc implementation](http://pandoc.org/MANUAL.html#definition-lists) can safely be considered to be chosen wisely, since it's author, John McFarlane, is also the maintainer of the CommonMark specification.

So IMHO **the next move is a no-brainer**:
just choose the same syntax as Pandoc's, and implement it as an extension, so as to bring back to the Rust community a means of expression that has existed for 50 years with troff, and since 1999 with HTML !

**Comment by marcusklaas** [comment]
It seems like this should be reasonably straightforward to implement as an extension. I can mentor anyone that would like to try to do this. Shoot me a message here or on the #pulldown-cmark channel at https://xi.zulipchat.com.

**Comment by ngirard** [comment]
@marcusklaas thanks for your mentoring offer... I'm in !

**Comment by shonfeder** [comment]
Hi! I just thought I'd check in here to see if this is still in the works, or if it may be stalled out and in need of more energy/support.

**Comment by shonfeder** [comment]
@marcusklaas I'd be happy to take a crack at it. If you have any initial pointers that you think might help, please share at your leisure. Otherwise I'll circle back here once I figure out enough to form any questions, or, optimistically, to suggest my proposed implementation strategy.

**Comment by marcusklaas** [comment]
That's awesome! I only have very general pointers: first try to get a general idea of the parsing strategy (two phases) and then think of where description lists could fit in. Starting with one or two basic examples and trying to get those to parse those first seems to be another good way to get started. Hit me up here or on zulip when you have questions!

**Comment by gdamore** [comment]
Here I am in 2024, and moving to mdbook from asciidoctor, and wishing for this.

**Comment by elijah-potter** [comment]
As a direct consumer of `pulldown_cmark` I think it is more important that v0.11 comes out sooner (with math support, sans description lists) than to wait and include both.

**Comment by kamoshi** [comment]
For the time being it's possible to use math like this:
```
pulldown-cmark = { git = "https://github.com/pulldown-cmark/pulldown-cmark.git", branch = "branch_0.11" }
```

**Comment by Martin1887** [comment]
Since the development has been slowed down a bit, the 0.11 has been released and description lists will be included in the 0.12 version. Thanks for your feedback.

**Comment by Martin1887** [comment]
Fixed by #915.
